docs/pages/machine-id/guides/github-actions-kubernetes.mdx

```diff
@@ -0,0 +1,221 @@
+---
+title: Using Machine ID With GitHub Actions and Kubernetes
+description: A tutorial for using Machine ID with GitHub Actions and Kubernetes
+---
+
+<Details
+  title="Version warning"
+  opened={true}
+  scope={["oss", "enterprise"]}
+  scopeOnly={true}
+  min="11.0"
+>
+  Machine ID for GitHub Actions is available starting from Teleport `v11.0`.
+</Details>
+
+GitHub Actions is a popular CI/CD platform that works as a part of the larger
+GitHub ecosystem.
+
+In this guide, we will use Teleport Machine ID and Kubernetes Access to allow a
+GitHub Actions workflow to securely connect to a Kubernetes cluster without the
+need for long-lived secrets.
+
+Teleport supports secure joining on both GitHub-hosted and self-hosted GitHub
+Actions runners as well as GitHub Enterprise Server.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A Kubernetes cluster connected to your Teleport cluster. If you do not already
```

Comment:
If you wanted, you could leverage `docs/pages/includes/tctl.mdx` to cover `tsh` and `tctl` login, and have a separate bullet point to cover what permissions your Teleport user needs.